src/webclients/components/sdfv/utils/attributes_table.ts

```diff
@@ -772,52 +895,74 @@ export function attributeTablePutEntry(
                     sdfgMetaDict['__reverse_type_lookup__'] &&
                     sdfgMetaDict['__reverse_type_lookup__'][valType])
                     valMeta = sdfgMetaDict['__reverse_type_lookup__'][valType];
+                const allowAdding = addDeleteButton;
                 attrTablePutDict(
                     key, undefined, val, elem, xform, target, valueCell, dtype,
-                    valMeta
+                    valMeta, allowAdding
                 );
                 break;
             case 'set':
             case 'list':
             case 'tuple':
                 let elemType = undefined;
-                let elemMety = undefined;
+                let elemMeta = undefined;
                 if (meta !== undefined && meta['element_type'])
                     elemType = meta['element_type'];
                 if (sdfgMetaDict && elemType &&
                     sdfgMetaDict['__reverse_type_lookup__'] &&
                     sdfgMetaDict['__reverse_type_lookup__'][elemType])
-                    elemMety =
+                    elemMeta =
                         sdfgMetaDict['__reverse_type_lookup__'][elemType];
-                valProp = attrTablePutList(
+                valProp = [attrTablePutList(
                     key, undefined, val, elem, xform, target, valueCell, dtype,
-                    elemMety
-                );
+                    elemMeta
+                )];
                 break;
             case 'Range':
             case 'SubsetProperty':
-                valProp = attrTablePutRange(
+                valProp = [attrTablePutRange(
                     key, undefined, val, elem, xform, target, valueCell, dtype
-                );
+                )];
                 break;
             case 'DataProperty':
-                valProp = attrTablePutSelect(
+                valProp = [attrTablePutSelect(
                     key, undefined, val, elem, xform, target, valueCell, dtype,
                     elem ? Object.keys(elem.sdfg.attributes._arrays): []
-                );
+                )];
                 break;
             case 'CodeBlock':
-                valProp = attrTablePutCode(
+                valProp = [attrTablePutCode(
                     key, undefined, val ? val.string_data : '', elem, xform,
                     target, valueCell, dtype
+                )];
+                break;
+            case 'Array':
```

Comment:
hmm, maybe we should have the type be "Data" in the json and subtype or something have the actual descriptor type

Reply:
I'm all for this, yes! The current approach relies on patching this switch statement when additional types are introduced in DaCe.